include/ginkgo/core/distributed/matrix.hpp

```diff
@@ -521,19 +523,27 @@ class Matrix
      *                 shared values is automatically extracted.
      * @return  MPI request for the non-blocking communication.
      */
-    mpi::request communicate(const local_vector_type* local_b) const;
+    [[deprecated(
+        "The pre-C++17 interface of the distributed::Matrix is "
+        "deprecated, please update your C++ standard to 17. ")]] mpi::request
+    communicate(const local_vector_type* local_b) const;
+#endif
```

Comment:
I think the `communicate` function should be the responsibility of this class rather than the sparse_communicator class.